src/algebra/big_operators/basic.lean

```diff
@@ -136,6 +136,7 @@ lemma monoid_hom.coe_prod [monoid β] [comm_monoid γ] (f : α → β →* γ) (
   ⇑(∏ x in s, f x) = ∏ x in s, f x :=
 (monoid_hom.coe_fn β γ).map_prod _ _
 
+-- This lemma is strictly weaker than `finset.prod_apply`, and should be removed.
```

Comment:
I don't see how this is the case `finset.prod_apply` states `(∏ x in s, f x) b`, but this states `coe_fn (∏ x in s, f x) b`

Reply:
Can't you get the conclusion of this lemma simply by forgetting that all the `f x` are `monoid_hom`s, and applying `finset.prod_apply`?

Reply:
That means you have to translate `coe_fn (∏ x in s, f x) b` to `(∏ x in s, coe_fn (f x)) b`, which is exactly what this lemma is doing

Reply:
But we seem to have a precedent for providing separate `X.coe_Y` and `X.Y_apply` lemmas even though the latter is usually provable by `rw [X.coe_Y, pi.Y_apply]`, for instance `linear_map.coe_add` and `linear_map.add_apply`.